traversal/iterator/GraphIterator.java

```diff
@@ -360,48 +354,55 @@ public Scoped get(Identifier.Variable scope) {
 
         public static class Scoped {
 
-            private final Set<ThingVertex> roleInstances;
-            private final PriorityQueue<Integer> edgeOrders;
+            Set<ThingVertex> roles;
+            TreeMap<Integer, ThingVertex> visited;
```

Comment:
a `TreeMap` can tell us the _last_ element in the collection. Since the `Integer` is the edge order, and we always walk the edges in order, getting the last element is equivalent to getting the last added element!

Deleting the last element it corresponds to popping from a stack, and adding to the treemap should only be allowed when the new element being added is greater than the last greater one, under the use case we have in the graph iterator

Reply:
the `set` is to allow fast lookups for what is contained in the scope already